apps/dashboard/src/components/activity/activity-table.tsx

```diff
@@ -0,0 +1,171 @@
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from '@/components/primitives/table';
+import { format } from 'date-fns';
+import { cn } from '@/utils/ui';
+import { IActivity, ISubscriber } from '@novu/shared';
+import { TimeDisplayHoverCard } from '@/components/time-display-hover-card';
+import { createSearchParams, useLocation, useSearchParams, useNavigate } from 'react-router-dom';
+import { useQueryClient } from '@tanstack/react-query';
+import { QueryKeys } from '@/utils/query-keys';
+import { useEnvironment } from '@/context/environment/hooks';
+import { getNotification } from '@/api/activity';
+import { useActivities } from '@/hooks/use-activities';
+import { StatusBadge } from './components/status-badge';
+import { StepIndicators } from './components/step-indicators';
+import { Pagination } from './components/pagination';
+import { useRef, useEffect } from 'react';
+
+export interface ActivityTableProps {
+  selectedActivityId: string | null;
+  onActivitySelect: (activity: IActivity) => void;
+}
+
+export function ActivityTable({ selectedActivityId, onActivitySelect }: ActivityTableProps) {
+  const queryClient = useQueryClient();
+  const { currentEnvironment } = useEnvironment();
+  const hoverTimerRef = useRef<NodeJS.Timeout | null>(null);
+  const [searchParams] = useSearchParams();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { activities, isLoading, hasMore } = useActivities();
+
+  const offset = parseInt(searchParams.get('offset') || '0');
+  const limit = parseInt(searchParams.get('limit') || '10');
+
+  const handleOffsetChange = (newOffset: number) => {
+    const newParams = createSearchParams({
+      ...Object.fromEntries(searchParams),
+      offset: newOffset.toString(),
+    });
+    navigate(`${location.pathname}?${newParams}`);
+  };
+
+  const handleRowMouseEnter = (activity: IActivity) => {
+    hoverTimerRef.current = setTimeout(() => {
+      queryClient.prefetchQuery({
+        queryKey: [QueryKeys.fetchActivity, currentEnvironment?._id, activity._id],
+        queryFn: () => getNotification(activity._id, currentEnvironment!),
+      });
+    }, 1000);
+  };
```

Comment:
I'm not sure if it's needed, but the goal is to preload the hovered pages to eliminate loading times

Reply:
I like the idea, but I am not sure if it's practical, as the activity page is usually visited with a specific transaction ID.

If we want to give it a shot, l'd also try prefetching all entries from the first list using resource hints and Hemnet (same strategy we did in a previous PR for images) so that the browser does the orchestration automatically, it uses its cache, etc...

Reply:
I think this can be mostly useful during the initial building and onboarding phases, where a user might be triggering something and then going to check it. I will leave it as I think it makes the page feel more snappy. Regarding using browser hints, I'm not sure if it will work as we don't use cookies and etc.. for auth.